restructuredtext_lint/sphinx.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import absolute_import
+
+try:
+    import sphinx  # noqa
+
+    # All currently know sphinx domains.
+    #
+    # See: https://github.com/sphinx-doc/sphinx/tree/1.3/sphinx/domains
+    from sphinx.domains import c as c_domain
+    from sphinx.domains import cpp as cpp_domain
+    from sphinx.domains import javascript as javascript_domain
+    from sphinx.domains import python as python_domain
+    from sphinx.domains import std as std_domain
+
+    SPHINX_AVAILABLE = True
+except ImportError:
+    SPHINX_AVAILABLE = False
+
+# Default and/or base roles/directives to ignore.
+#
+# See: http://sphinx-doc.org/markup/para.html
+# And: http://sphinx-doc.org/markup/toctree.html#directive-toctree
+# And: https://github.com/sphinx-doc/sphinx/blob/1.3/sphinx/directives/other.py
+_base_sp_roles = ('ctype',)
+_base_sp_directives = ('autosummary', 'centered', 'currentmodule',
+                       'deprecated', 'hlist', 'include', 'index',
+                       'literalinclude', 'no-code-block', 'seealso',
+                       'toctree', 'todo', 'versionadded', 'versionchanged')
+
+
+def fetch_ignore_roles_directives():
+    """Extract all possible directives & roles that sphinx is aware of.
+
+    Raises a ``RuntimeError`` if sphinx is not importable.
+
+    :rtype dict: dict with 'directives' and 'roles' keys with list values.
+    """
+    if not SPHINX_AVAILABLE:
+        raise RuntimeError("Sphinx roles & directives can not be fetched"
+                           " without sphinx being importable")
```

Comment:
This error message is placing blame on the user. That is always a bad idea because it makes the user defensive and can lead to aggressive issues. Instead, let's try some wording like:

```
'`restructuredtext-lint` tried to `import sphinx` at the initial load time'
'but was unable to. Please verify `sphinx` is installed properly.'
```